### ISSUE TITLE
SVG doesn't load in browser

### ISSUE THREAD

**Issue by carmennnnandra** [jonasschmedtmann/complete-javascript-course#157]
Hello, Jonas!
On forkify application, svg doesn t appear on browser
Can you help me ?
Thank you

**Comment by Safeeerrr** [comment]
Yes i am having the same issue. Install parcel beta 2 version which is the latest it. After building with parcel only the svg icons of the header are not rendered...couldn't figure out the issue. 

![forkify](https://user-images.githubusercontent.com/20049457/111558002-21f8ee80-878e-11eb-8c2d-c8e73dedf203.JPG)

**Comment by carmennnnandra** [comment]
> Yes i am having the same issue. Install parcel beta 2 version which is the latest it. After building with parcel only the svg icons of the header are not rendered...couldn't figure out the issue.
> 
> ![forkify](https://user-images.githubusercontent.com/20049457/111558002-21f8ee80-878e-11eb-8c2d-c8e73dedf203.JPG)

And you downloaded the same icons.vg like Jonas, right?

**Comment by banicaDaniel07** [comment]
Got the same problem, try to:
-reinstall parcel, I have v2
-rewrite this import icons from 'url:../img/icons.svg';
-restart the textEditor, pc, npm start etc
for me worked after reinstall. Hope it helps

**Comment by carmennnnandra** [comment]
now i have this error '@parcel/transformer-js: This experimental syntax requires enabling one of the following parser plugin(s): 'classPrivateProperties, classPrivateMethods' (2:4) '
and i didn't solve


this is my json 
{
  "name": "licenta_carmen",
  "version": "1.0.0",
  "description": "Recipe application",
  "main": "index.html",
  "scripts": {
    "start": "parcel index.html",
    "build": "parcel build index.html"
  },
  "author": "Melniciuc Carmen Andra",
  "license": "ISC",
  "devDependencies": {
    "@babel/plugin-proposal-private-methods": "^7.13.0",
    "@parcel/transformer-sass": "^2.0.0-beta.2",
    "parcel": "^2.0.0-beta.2"
  },
  "dependencies": {
    "core-js": "^3.9.1",
    "regenerator-runtime": "^0.13.7"
  }
  
}

**Comment by AmirVahedix** [comment]
I have the same issue using parcel 2.0.0beta2
it loads them properly but the browser can't show them :(
I tried different browsers but any of them worked :(

**Comment by carmennnnandra** [comment]
> Yes i am having the same issue. Install parcel beta 2 version which is the latest it. After building with parcel only the svg icons of the header are not rendered...couldn't figure out the issue.
> 
> ![forkify](https://user-images.githubusercontent.com/20049457/111558002-21f8ee80-878e-11eb-8c2d-c8e73dedf203.JPG)

how did you change the colors? because i m changing too but in browser doesn t change anything

**Comment by SStranks** [comment]
I had the same issue, using Parcel 2.0.0beta2. The icons in the top banner did not render, yet they appear to be there if you inspect the page (image attached). They rendered in the recipe list fine though. 
![Icon Error](https://user-images.githubusercontent.com/79522719/119043095-9596c200-b9b0-11eb-9697-8fc2a125f9e1.png)

**Comment by pranjalg131** [comment]
Any Fix till now ?

**Comment by pranjalg131** [comment]
> Any Fix till now ?

Just use the version same to jonas by doing
npm uninstall parcel 
npm install parcel@2.0.0-beta.1
npm start 

It worked for me

**Comment by Alieltahan** [comment]
> > Any Fix till now ?
> 
> Just use the version same to jonas by doing
> npm uninstall parcel
> npm install parcel@2.0.0-beta.1
> npm start
> 
> It worked for me

This solution worked

**Comment by Felixmurithi** [comment]
> Just use the version same to jonas by doing npm uninstall parcel npm install parcel@2.0.0-beta.1 npm start
> 
> It worked for me





> > > Any Fix till now ?
> > 
> > 
> > Just use the version same to jonas by doing
> > npm uninstall parcel
> > npm install parcel@2.0.0-beta.1
> > npm start
> > It worked for me
> 
> This solution worked.
> 
> While building the app, in case faced error: `index.html: tree.render is not a function` add **.htmlnanorc.js** to project root with the following: `module.exports = { "minifySvg": false }`

to fix 
`index.html: tree.render is not a function`
add a file named 
`18-forkify\.htmlnanorc`
and add 
```
{
  "minifySvg": false
}
```

**Comment by Harsh-Indian** [comment]
hey can anyone help me ? i got same error as i deployed my forkify application on natlfiy

**Comment by shubhs27** [comment]
I have the same error, is there a fix available

**Comment by jmgates12a** [comment]
Use **const icons = require('url:../../img/icons.svg');** instead of **import icons from 'url:../../img/icons.svg';** worked for me when I had this problem at the developement stage but not when building via Netlify.

**Comment by Chinmay-G** [comment]
> Use **const icons = require('url:../../img/icons.svg');** instead of **import icons from 'url:../../img/icons.svg';** worked for me

Heyy, which parcel version did you use?

**Comment by jmgates12a** [comment]
Sorry I'm very much a newby and I was experiencing the problem of svg icons
not appearing on the app when I built it in development. Having  made the
change above it appeared to resolve the issue but only in development. The
problem has since re-emerged when building the app through Netlify.
Apologies for the misleading post. If I get a resolution I'll enter another
post.

**Comment by Chinmay-G** [comment]
> Sorry I'm very much a newby and I was experiencing the problem of svg icons
> not appearing on the app when I built it in development. Having  made the
> change above it appeared to resolve the issue but only in development. The
> problem has since re-emerged when building the app through Netlify.
> Apologies for the misleading post. If I get a resolution I'll enter another
> post.
> […](#)

No problem man.. and Hey, I found a solution.

**Comment by Chinmay-G** [comment]
> Sorry I'm very much a newby and I was experiencing the problem of svg icons
> not appearing on the app when I built it in development. Having  made the
> change above it appeared to resolve the issue but only in development. The
> problem has since re-emerged when building the app through Netlify.
> Apologies for the misleading post. If I get a resolution I'll enter another
> post.
> […](#)

No problem man. And hey, I found a solution,
add  --no-minify to your build command in Netlify build settings.
Should prbly look like this -
parcel build index.html --dist-dir ./dist --no-minify

**Comment by slydevruby** [comment]
Parcel cannot get the proper svg file in 'dist' directory, it is empty.
I used 'vite' instead of 'parcel' eventually.

package.json :
```
  "scripts": {
    "dev": "vite",
    "build": "vite build",
    "lint": "npx eslint $1"
  },
```
and instead of 
`import icons from 'url:../../img/icons.svg'; `
I used
`import icons from '../../img/icons.svg'; `

**Comment by jmgates12a** [comment]
Ok, I'll check it out.

Thanks for that.

**Comment by vagman** [comment]
> Parcel cannot get the proper svg file in 'dist' directory, it is empty. I used 'vite' instead of 'parcel' eventually.
> 
> package.json :
> 
> ```
>   "scripts": {
>     "dev": "vite",
>     "build": "vite build",
>     "lint": "npx eslint $1"
>   },
> ```
> 
> and instead of `import icons from 'url:../../img/icons.svg'; ` I used `import icons from '../../img/icons.svg'; `

Migration from Parcel to Vite worked fine. Thank you!
